app/components/detection/camera.py

```diff
@@ -0,0 +1,235 @@
+
+        if filters_config.spatial.enabled:
+            self.spatial = rs.spatial_filter()
+            # TODO: Might be needed
+            # self.spatial.set_option(rs.option.filter_magnitude, 2)
+            # self.spatial.set_option(rs.option.filter_smooth_alpha, 0.5)
+            # self.spatial.set_option(rs.option.filter_smooth_delta, 20)
+
+        if filters_config.temporal.enabled:
+            self.temporal = rs.temporal_filter()
+
+        if filters_config.hole_filling.enabled:
+            self.hole_filling = rs.hole_filling_filter()
+
+        logger.debug(f"Filters initialized: Spatial={bool(self.spatial)}, Temporal={bool(self.temporal)}", operation="init_pipeline")
+
+    def _initialize_pipeline(self):
+        """Initialize the RealSense pipeline with proper reset sequence."""
+        logger.debug("Initializing RealSense pipeline sequence", operation="init_pipeline")
+
+        ctx = rs.context()
+        devices = ctx.query_devices()
+        if not devices:
+            raise RuntimeError("Lost device after reset sequence.")
+    
+        # Configure Pipeline
+        self.pipeline = rs.pipeline()
+        rs_config = rs.config()
+
+        logger.debug(
+            f"Configuring streams: {self.width}x{self.height} @ {self.fps}fps",
+            operation="init_pipeline",
+        )
+        rs_config.enable_stream(
+            rs.stream.color, self.width, self.height, rs.format.bgr8, self.fps
+        )
+        rs_config.enable_stream(
+            rs.stream.depth, self.width, self.height, rs.format.z16, self.fps
+        )
+
+        # Initialize helper objects
+        self.align = rs.align(rs.stream.color)
+        self._setup_filters()
+        
+        # Start Pipeline
+        logger.debug("Starting RealSense pipeline...", operation="init_pipeline")
+        try:
+            profile = self.pipeline.start(rs_config)
+
+            time.sleep(1) # FUCKKKKKK
+
+            depth_sensor = profile.get_device().first_depth_sensor()
+            try:
+                depth_sensor.set_option(rs.option.visual_preset, 1)
+                depth_sensor.set_option(rs.option.laser_power, 360)
+            except Exception as e:
+                logger.error(f"Exception while loading preset config onto camera: {e}", exc_info=True)
+
+            logger.info("Loading preset config onto camera...")
+            
+            # Warmup: Discard first few frames to allow auto-exposure to settle
+            for _ in range(5):
+                self.pipeline.wait_for_frames(timeout_ms=1000)
+                
+            logger.info("RealSense pipeline started and warmed up.", operation="init_pipeline", status="success")
+            
+        except RuntimeError as e:
+            if "device is busy" in str(e).lower():
+                logger.error("Device Busy: Check if another process is using the camera or if USB cable is loose.")
+            raise e
+
+    def on_iteration(self):
```

Comment:
```suggestion
    def loop(self):
```
Or stick with `iterate`.